### PR TITLE
Add support for Node 6 & 7

### PR DESCRIPTION
**Pull request by mcampa**
This uses the new `@bigcommerce/stencil-styles#1.0.6` which uses a [fork](https://github.com/bigcommerce-labs/node-sass) of `node-sass` compatible with node 6 & 7

Plus, adding AppVeyor for windows tests

@bigcommerce/stencil-team

### CODE REVIEW

File: lib/theme-config.spec.js [hunk 1]
```diff
@@ -1,40 +1,42 @@
...
-        it ('should allow overwriting of configPath, schemaPath, and variationName by calling getInstance with params', function(done) {
-            var secondConfigPath = '/fake/config.json';
-            var secondSchemaPath = '/fake/schema.json';
-            var secondVariationName = 'Second';
-            var secondThemeConfig;
-
-            secondThemeConfig = ThemeConfig.getInstance('/fake');
+        it ('should allow overwriting of configPath, schemaPath, and variationName by calling getInstance with params', done => {
+            const isWindows = Os.platform() === 'win32';
+            const secondConfigPath = isWindows ? '\\fake\\config.json' : '/fake/config.json';
```
Comment (mjschock) [inline lib/theme-config.spec.js]: i think we use this [path.posix method]|(https://nodejs.org/api/path.html#path_path_posix) instead.  from [this page](https://nodejs.org/api/path.html#path_windows_vs_posix) i think we could do:

```
const normalize = path.posix.normalize;
expect(normalize(secondThemeConfig.configPath)).to.equal(normalize(secondConfigPath));
```

Comment (mcampa) [inline lib/theme-config.spec.js]: awesome

**Comment by mjschock** [comment]
LGTM 👍

File: lib/theme-config.spec.js (hunk 1 above)
Comment (mcampa) [inline lib/theme-config.spec.js]: doesn't work :( reverted

**Comment by adammcarth** [comment]
Thanks @mcampa 👍

**Pull request merged**